RequestsPlus/RequestsPlus.py

```diff
@@ -0,0 +1,150 @@
+import requests
+import urlnorm
+
+import util
+from .ResponseCollection import *
+
+HTTP_HEADER_AUTHORIZATION = 'Authorization'
+AUTHZ_TYPE_BEARER = 'Bearer'
+MIME_TYPE_JSON = 'application/json'
+HTTP_HEADER_CONTENT_TYPE = 'Content-type'
+
+
+class RequestsPlus(util.UtilMixin, object):
+    def __init__(self, apiBaseURL, contentType=MIME_TYPE_JSON, authZToken=None, authZType=AUTHZ_TYPE_BEARER):
+        self._name = self.__class__.__name__
+        self.apiBaseURL = apiBaseURL
+        self.contentType = contentType
+        self.authZToken = authZToken
+        self.authZType = authZType
+        self.session = requests.Session()
+        self.session.headers.update(self._prepareHeaders())
+
+    @staticmethod
+    def responseCollection(response):
+        """
+        Convenience method to make a ResponseCollection
+        object for a response.
+
+        :param response: requests Response object, usually the first of multiple pages
+        :type response: requests.models.Response
+        :return: ResponseCollection object containing multiple response pages
+        :rtype: ResponseCollection
+        """
+        return ResponseCollection(response)
+
+    @property
+    def _authZHeader(self):
+        return {} if self.authZToken is None else \
+            {HTTP_HEADER_AUTHORIZATION: ' '.join([self.authZType, self.authZToken + ''])}
+
+    @property
+    def _contentTypeHeader(self):
+        return {} if self.contentType is None else \
+            {HTTP_HEADER_CONTENT_TYPE: self.contentType}
+
+    def _prepareHeaders(self):
+        headers = {}
+        headers.update(self._contentTypeHeader)
+        headers.update(self._authZHeader)
+        return headers
+
+    def _prepareURL(self, apiQueryURI):
+        """
+        If the URI (actually just a partial URL, usually the path part) doesn't begin with
+        the base URL for the API, concatenate the two into a new URL and return it.
+
+        :param apiQueryURI: URI (actually, just a partial URL, usually the path part) for an API entry point.
+        :type apiQueryURI: str
+        :return: URL for the API query, ready for use
+        :rtype: str
+        """
+        assert isinstance(apiQueryURI, str)
+        assert not util.stringContainsAllCharacters(apiQueryURI, '{}'), \
+            'apiQueryURI contains unformatted arguments: "%s"' % apiQueryURI
+
+        if apiQueryURI.startswith(self.apiBaseURL):
+            return apiQueryURI
+        else:
```

Comment:
True enough!  I usually catch this kind of thing myself.  That proves that another set of eyes on the code is valuable.